lib/surgex/parser/parsers/id_list_parser.ex

```diff
@@ -46,7 +46,7 @@ defmodule Surgex.Parser.IdListParser do
     {:ok, [id | previous_ids]}
   end
 
-  defp reduce_ids(id, {:ok, _previous_ids}) when is_integer(id) do
+  defp reduce_ids(_id, {:ok, _previous_ids}) do
```

Comment:
I decided to treat "anything else" as `:invalid_identifier` but we could also handle `nil` specifically instead:

```
defp reduce_ids(nil, {:ok, _previous_ids}) do
  {:error, :invalid_identifier}
end
```

But I think the more general approach is more robust.

Reply:
Agree, more generic approach makes more sense.